pykern/pkcompat.py

```diff
@@ -99,3 +114,35 @@ def _assert_type(value, typ):
                 type(value),
             ),
         )
+
+
+def _version_ok(major, minor):
+    m = [int(i) for i in platform.python_version_tuple()]
+    if m[0] > major:
+        return True
+    return m[0] == major and m[1] >= minor
+
+
+def _zip_strict(iterables):
+    """Code from https://stackoverflow.com/a/69485272"""
+
+    def _first_tail():
+        nonlocal first_stopped
+        first_stopped = True
+        return
+        yield
```

Comment:
How is this reached if we return before?

Reply:
Yield forces it to be a coroutine (generator).

Reply:
Oh very interesting. Thanks.